js/Core/Chart/StockChart.js

```diff
@@ -699,9 +699,11 @@ addEvent(Series, 'render', function () {
                 clipHeight -= Math.max(lineHeightCorrection - dist, 0);
             }
         }
-        // First render, initial clip box
-        if (!this.clipBox && this.isDirty && !this.isDirtyData) {
-            this.clipBox = merge(chart.clipBox);
+        // First render, initial clip box. clipBox also needs to be updated if
+        // the series is rendered again before starting animating, in
+        // compliance with a responsive rule (#13858).
+        if (!chart.hasLoaded || (!this.clipBox && this.isDirty && !this.isDirtyData)) {
+            this.clipBox = this.clipBox || merge(chart.clipBox);
```

Comment:
Is there any reason to use the original object, then the copy (if not defined)?

Reply:
Just to avoid the copy. The demos with visual differences dont even load the code that got changed by this PR, so not sure whats up there.

Reply:
Just realized it's `this.clipBox` vs `chart.clipBox`, sorry!

Regarding the visual diff, perhaps we load all modules in CircleCi visual tests? Just like in Karma. @goransle might have more info :)

Reply:
Yes, in fact the visual tests are a subset of the karma tests 🤓

Reply:
So I was kind of able to reproduce the staticscale one by loading stock, but its reversed from what the visual diff says, broken on master and fixed in this PR, strange.

Master: https://jsfiddle.net/qp18r2xb/
PR: https://jsfiddle.net/wzfvnp1t/12/

Reply:
@khlieng you might want to run karma in debug mode to see what is happening there. 
From the highcharts root: `npx karma start ./test/karma-conf.js --debug --browsers Chrome --no-single-run --tests 'highcharts/studies/axis-staticscale'`

Reply:
Now the staticscale diff is working as intended, but im not able to reproduce the other one.

That demo also seems to be broken, the scrollable part doesnt work, and try this:
1. Open https://www.highcharts.com/samples/gantt/chart/scrollable-plotarea-vertical
2. Click "New product launch once"
3. Click "New offices" 3 times

Reply:
Okaay, some magic happens here. If this is not already reported could you report it? (Gantt+scrollable+tree grid axis)

Reply:
#13884 looks similar, and it has an open and approved PR 🤔

Reply:
I added the fix from that PR here just to test and now the clipping in the diff is fine.

Reply:
There are still two diffing samples? https://vrevs.highsoft.com/pr/15200/review

Reply:
The first one is working as intended, it now has the same clipping as a regular bar chart and the same clipping as that sample without stock loaded: https://www.highcharts.com/samples/highcharts/studies/axis-staticscale.

And the other one is the same 15px diff as in #13884, not sure why that happens, but the 600px width it has now kind of sounds more correct.